File: Python/Kode Program/Dokter.py
from Manusia import Manusia

class Dokter(Manusia):
    def __init__(self, NIK, nama, umur, jenisKelamin, spesialis, jadwal):
        super().__init__(NIK, nama, umur, jenisKelamin)
        self.spesialis = spesialis
        self.jadwal = jadwal
    
    def getSpesialis(self):
        return self.spesialis
    
    def setSpesialis(self, spesialis):
        self.spesialis = spesialis
    
    def getJadwal(self):
        return self.jadwal
    
    def setJadwal(self, jadwal):
        self.jadwal = jadwal

    def printDokter(self):
        print(f"NIK : {self.getNIK()}, "
              f"Nama : {self.getNama()}, "
              f"Umur : {self.getUmur()}, "
              f"Jenis Kelamin : {self.getJenisKelamin()}, "
              f"Spesialisasi : {self.getSpesialis()}, "
              f"Jadwal : {self.getJadwal()}")

File: CPP/Kode Program/Dokter.cpp
#include <iostream>
#include <vector>
#include "Manusia.cpp"

using namespace std;

//Class Dokter extends Class Manusia
class Dokter : public Manusia{
    //Atribut Class Dokter
    private:
        string spesialis;
        string jadwal;

    public:
        //Constructor
        Dokter(int NIK, string nama, string umur, string jenisKelamin, string spesialis, string jadwal) : Manusia(NIK, nama, umur, jenisKelamin){
            this->spesialis = spesialis;
            this->jadwal = jadwal;
        }

        //Getter dan Setter
        string getSpesialis()const{
            return spesialis;
        }

        void setSpesialis(string spesialis){
            this->spesialis = spesialis;
        }

        string getJadwal()const{
            return jadwal;
        }

        void setJadwal(string jadwal){
            this->jadwal = jadwal;
        }

        //Cetak info / data dari seorang dokter
        void printDokter(){
            cout << "NIK : " << getNIK()
            << ", Nama : " << getNama()
            << ", Umur : " << getUmur()
            << ", Jenis Kelamin : " << getJenisKelamin()
            << ", Spesialisasi : " << getSpesialis()
            << ", Jadwal : " << getJadwal() << endl;
        }

        ~Dokter(){
        }
};